src/lib/logger.js

```diff
@@ -10,9 +10,9 @@ module.exports = {
   },
   error(...args) {
     const [format, ...rest] = args;
-    console.error(
-      `${chalk.grey('[Semantic release]:')} ${chalk.red(format instanceof Error ? format.stack : format)}`,
-      ...rest.map(arg => chalk.red(arg instanceof Error ? arg.stack : arg))
-    );
+    console.error(`${chalk.grey('[Semantic release]:')} ${chalk.red(format)}`, ...rest.map(arg => chalk.yellow(arg)));
+    args.filter(arg => arg instanceof Error).forEach(error => {
+      console.error(error);
```

Comment:
since you use `%s`, would this not print the error twice? Instead of this handling, why not simply log like this:
```js
logger.error('An error occurred while running semantic-release: ', err)
```
(no `%s`)

Reply:
The function `logger.error` can be used to print anything, not only `Error` objects. So it has to work with string formatting and handled multiple parameters.

The function format the the parameters and print them in red, and in addition if there is errors among the parameters it will print them completely.

So the error message will appear in the formatted string and the complete error (with message, stacktrace and properties) will be logged after.

Reply:
`console.log`/`console.error` support format strings and arbitrary structures which result in being `util.inspect()`ed. It behaves differently depending on whether the first argument is a format string or not, which changes the semantic of the rest arguments.
This logger seems to only support the format string mode, and I assume `chalk.yellow()` implicitly converts the argument to a string ala `toString()`, so for arbitrary structures it would just print `[object Object]`.
I would make it behave like the native `console` as in not add custom handling of errors, but of format strings if that is needed.

Reply:
I made some changes. The Object will now be logged properly.
Both `log` and `error` support the following now:
- `logger.log(obj)`
- `logger.log('Something %O something else', obj)`
- `logger.log('Something %s something else','some string')`
- `logger.log('Something %s something else %O','some string', obj)`